### ISSUE TITLE
Do I need the kalifs.zip at all? 

### ISSUE THREAD

**Issue by noyb666** [offensive-security/kali-nethunter#1199]
### Device:
angler 6p

### OS version (KitKat/Lollipop/Marshmallow/Nougat):
oreo

### Built from repo (date and build command) or downloaded from website (links):
oreo nightly

## Issue:
I posted in the NRT thread, but then realized that here might be a better place... So what's the deal with this kalifs.zip, that I had to download in step 2 and that was never mentioned again...? 

Here the full description:

So I followed the install guide step by step and that brought me to this
site (offensive-security/nethunter-LRT) and I continued to follow all the steps.

Everything looked totally fine, the terminal output ends with:

"Rebooting into Kali Linux Nethunter

Everything is installed. Phone should be booting up! Enjoy Nethunter
"

Great, I thought, but the phone rebooted and nothing happened. I ended up with a black screen and a white "Google" writing and sadly that's already the end of the story, no Nethunter in sight!?

Any clue what went wrong and what the state of the phone is?

And then I also remembered that on the previous site that brought me here
(https://github.com/offensive-security/kali-nethunter/wiki) there in the second step
(2.0 Downloading NetHunter) it says:

"For a fresh install, you will need a nethunter-generic-[arch]-kalifs-.zip as well as a kernel-nethunter-[device]-[os]-.zip. The kernel should be flashed last. The update-nethunter-generic-[arch]-*.zip files are for updating your installation or if you wish to download the Kali rootfs inside the NetHunter app instead."

Ok, I downloaded all of it in that step, but now I realize that I never used them in the following steps, simply because they were not mentioned anymore....!?

Why have I downloaded the "kalifs".zip at all? When there's no step explanation what to do with it. So have I terribly missed something? Or can we call this install guide inconsistent and confusing?

And once again, why do I have the output saying everything is fine but nothing happens?

**Comment by ajes** [comment]
Nethunter require:
kernel-nethunter-*.zip - which include modified kernel, so some hacks like wifi will work
nethunter-generic-*-kalifs-*.zip - as this is full Kali system. Without it, you have not installed Nethunter :)

**Comment by yesimxev** [comment]
Yes kalifs contains the base system. You have to install nethunter-kernel too from eg. build.nethunter.com/nightly. You can also build fresh full installer (which has the fs, apps, kernel, su) using the nethunter-installer from github. You can add that to NRT too if you don't want to TWRP

**Comment by 0E800** [comment]
If you dont know what Kali file system is for I'm not sure anybody should be helping you arm your phone as a weapon that might ultimately wind you up in jail for not knowing how to use it or that in special circumstances  by having it you can be profiled as a potential cyber threat actor or worse.

There is a reason why it takes extreme effort to successfully port your device and even when you did everything correct; and everything seems to be working you still fall into problems and everybody wants you to show them how you did it and become irritated that their results don't match yours.

Do yourself a favor and install Kali and get yourself a few tutorials and when you get bored or stuck on that, alternate to getting familiar with installing custom roms and bricking and unbricking your device.

**Comment by noyb666** [comment]
@0E800 

While what you're saying is partially true it still sounds like a smartass comment and you're missing the point....

The point is, that this "install guide" (that doesn't even deserve this name...) is complete rubbish...!

So Kalifs is the most important file... Great. 
How come it's mentioned exactly once...!??? And then never again?

""For a fresh install, you will need (to download) a nethunter-generic-[arch]-kalifs-.zip..."

And now? Use your female intuition?

Why not, just don't call it a Install guide, wiki or documentation if you don't tell people what to do after the download....

But even worse:

this "NetHunter Linux Root Toolkit" won't work anyway, because it's outdated lousy crap and SuperSu will brick your device with 100% reliability...

The link provided brings you to a 3 year old outdated post....
And that's almost good news, because SuperSu is such a nonsense, that you should really look for alternatives.

And then there's this sentence:

"The kernel should be flashed last."

Again, no instruction / explanation follows...

If the "preferred method for installing NetHunter" is the "NetHunter Linux Root Toolkit " as the creators claim... when and how do you flash it "last"....?

Actually you supposed to: "Put the NetHunter zip image you downloaded or built in the /kaliNethunter folder."

Well ok, but explain how you flash the kernel last, when you just dump it in that folder? Together with the Kalifs....? Or not? Well.... This "guide" keeps it a secret.

Bottom line: Either have a documentation and install instructions that work, others can follow and understand or not....!

What we have here is an outdated confusing pile of rubbish that needs to be cleaned up and replaced with a SIMPLE and WORKING guide in order to not waste peoples time.

And the saddest thing is, that this guide not only doesn't work, it makes rather easy things unnecessarily difficult.

When Lineage OS can have a simple, logical and easy to follow guide, why can't Nethunter?

And actually I managed to flash it  eventually giving this "preferred method for installing NetHunter" a miss and following the Lineage guide and that works.

**Comment by yesimxev** [comment]
There are several ways but either LRT or flashing from TWRP, the process is same. LineageOS 15 needs AddonSU instead and apks need to be installed manually

**Comment by noyb666** [comment]
Yes, there are several ways... on top of the ones you named you can also install the APK's described here: https://forum.xda-developers.com/android/general/kali-nethunter-android-device-t3596625

But the point is: it would be nice to have a understandable docu, wiki, install guide... you name it.
Instead we are presented this crap here, that isn't even consistent about the supported devices to begin with... Then it "forgets" to tell you what to do with the Kalifs and when "later" is to flash the kernel and after innumerous hours and weekend sessions until 2 am you find out that this LRT  (" our preferred method for installing NetHunter" yeah, my ass....) won't work at all because it's complete useless  outdated nonsense....
(To be be more precise: The first part works (flashing the stock image) but not the second, SuperSu will brick it and it's unclear what exactly to put in the " /kaliNethunter folder.")

So the point is...: Apparently different people have long time ago written this tutorial and never ever read, updated or corrected it and now we're looking at this mess.
Honestly, without this "guide" I would have saved a lot of time and flashing the Kalifs couldn't be easier and more straightforward, what I fail is to flash a kernel that works, not only because it's still unclear what  "The kernel should be flashed last." is supposed to mean but probably also because I want to use it on a Lineage OS and there isn't a kernel for this OS around.

**Comment by amzker** [comment]
You All Making It Confusing

Kalifs.zip Fully  Required For Nethunter

IF Wanted To Install Just Full Kali Linux

Need Kali.img (2GB)

**Comment by yesimxev** [comment]
I know what you mean. Think like - if it's too obvious, it's easy enough for dangerous script kiddies :)
